Change and refactor routing

Moriartymath opened the pull request.

Moriartymath committed:
Add Women route with loader, Refactor code
- Add method of getting all female products to ProductAPI

Moriartymath committed:
Add route with loader for children

Moriartymath committed:
Refactor DetailProduct

Moriartymath committed:
Change routing
- Add layout to men women and children by placing them as nested routes of ProductsLayout ( products path)
- Create Products component instead of separate for each category (men, women , childre)
- Create loaders file, with loaders for each gender ( products)

Moriartymath committed:
Add conditional rendering to ProductsLayout to display separate product

Moriartymath committed:
Add Back button for product

Moriartymath committed:
Create component to hide filters

TarasBlatnoi reviewed (changes requested):
Fix typescript errors and add back previous realization you might like also products with reusing of nested loader and useRouteLoaderData

Moriartymath committed:
Remove Artificial route 'products'

Moriartymath committed:
change href in navBar

Moriartymath committed:
Change routing for products routes

- Change routing for women men and children by wraping them ( men and men/:id and the same for women and children) into Route with corresponding id and loader to share the same data between nested routes

Moriartymath committed:
Fix types warnings

TarasBlatnoi reviewed (approved):
LGTM
